src/shogun/io/TBOutputFormat.cpp

```diff
@@ -106,6 +107,13 @@ tensorflow::Event TBOutputFormat::convert_scalar(
 		    "Unsupported type %s", value.first.get_value().type_info().name());
 	}
 
+	auto write_summary = [&summaryValue=summaryValue, &value](auto type) {
+		summaryValue->set_simple_value(
+		    any_cast<decltype(type)>(value.first.get_value()));
+	};
+
+	for_each_type(value.first.get_value(), all_types, write_summary);
```

Comment:
sorry one more thing. Could we make the naming in a way that it is clear that this is a shogun utility function.
Like adding a prefix `sg_` to all functions and global variances that have to do with this

Reply:
OK! Btw, do you have any rule when to put or leave out functions/typedefs/enums from the namespace?

Reply:
minimal evasion, so only make something as visible/complex as needed, not more